railties/lib/rails/tasks/framework.rake

```diff
@@ -49,6 +49,7 @@ namespace :app do
     # desc "Adds new executables to the application bin/ directory"
     task :bin do
       Rails::AppUpdater.invoke_from_app_generator :update_bin_files
```

Comment:
Would not this still remove the spring to later add it back? I think it is better to have a different generator task that run spring binstubs and in case spring is skipped we just copy the content of the bin files.

Reply:
Sorry, I couldn't get what you mean.
Having a different generator task does not solve the problem since `app:update` cannot know whether spring is skipped or not because it cannnot accept option.
Or what you have in mind is too big to me maybe..

Now I'm inclined to just output warning after `app:update`:
```diff
diff --git a/railties/lib/rails/generators/rails/app/app_generator.rb b/railties/lib/rails/generators/rails/app/app_generator.rb
index 395ac7ef2f..b2539d738a 100644
--- a/railties/lib/rails/generators/rails/app/app_generator.rb
+++ b/railties/lib/rails/generators/rails/app/app_generator.rb
@@ -324,6 +324,7 @@ def create_credentials
       end

       def display_upgrade_guide_info
+        say "\nIf you want to keep spring working, please run `bundle exec spring binstub --all`"
         say "\nAfter this, check Rails upgrade guide at http://guides.rubyonrails.org/upgrading_ruby_on_rails.html for more details about upgrading your app."
       end
       remove_task :display_upgrade_guide_info
```

Reply:
What about something like this (from `bin/spring`):

```ruby
if defined?(Spring)
  # ...
end
```

or, this ugly begin-rescue (silently fail) (from `bin/rails`):

```ruby
begin
  load File.expand_path('../spring', __FILE__)
rescue LoadError => e
  raise unless e.message.include?('spring')
end
```

Maybe another solution would be to let `AppUpdater` accept cli options?

Reply:
Since #32837 is merged, we can use `options[:skip_spring]`. I'll revisit to address this later.